samples/custom-hostname-instance/createInstanceWithCustomHostname.js

```diff
@@ -0,0 +1,118 @@
+ *    $ gcloud compute images list
+ * @param {string} networkName - Name of the network you want the new instance to use.
+ *    For example: global/networks/default - if you want to use the default network.
+ */
+function main(
+  projectId,
+  zone,
+  instanceName,
+  hostname,
+  machineType = 'n1-standard-1',
+  sourceImage = 'projects/debian-cloud/global/images/family/debian-10',
+  networkName = 'global/networks/default'
+) {
+  // [START compute_instances_create_custom_hostname]
+  /**
+   * TODO(developer): Uncomment and replace these variables before running the sample.
+   */
+  // const projectId = 'YOUR_PROJECT_ID';
+  // const zone = 'europe-central2-b'
+  // const instanceName = 'YOUR_INSTANCE_NAME'
+  // const hostname = 'host.example.com'
+  // const machineType = 'n1-standard-1';
+  // const sourceImage = 'projects/debian-cloud/global/images/family/debian-10';
+  // const networkName = 'global/networks/default';
+
+  const compute = require('@google-cloud/compute');
+  const computeProtos = compute.protos.google.cloud.compute.v1;
+
+  // Create a new instance with the values provided above in the specified project and zone.
+  async function createInstanceWithCustomHostname() {
+    const instancesClient = new compute.InstancesClient();
+
+    console.log(
+      `Creating the ${instanceName} instance in ${zone} with hostname ${hostname}...`
+    );
+
+    const [response] = await instancesClient.insert({
+      instanceResource: {
+        name: instanceName,
+        // Custom hostnames are not resolved by the automatically created records
+        // provided by Compute Engine internal DNS.
+        // You must manually configure the DNS record for your custom hostname.
+        hostname,
+        disks: [
+          {
+            // Describe the size and source image of the boot disk to attach to the instance.
+            initializeParams: {
+              diskSizeGb: '10',
+              sourceImage,
+            },
+            autoDelete: true,
+            boot: true,
+            type: computeProtos.AttachedDisk.Type.PERSISTENT,
+          },
+        ],
+        machineType: `zones/${zone}/machineTypes/${machineType}`,
+        networkInterfaces: [
+          {
+            // Use the network interface provided in the networkName argument.
+            name: networkName,
+          },
+        ],
+      },
+      project: projectId,
+      zone,
+    });
+    let operation = response.latestResponse;
+    const operationsClient = new compute.ZoneOperationsClient();
+
+    // Wait for the create operation to complete.
+    while (operation.status !== 'DONE') {
+      [operation] = await operationsClient.wait({
+        operation: operation.name,
+        project: projectId,
+        zone: operation.zone.split('/').pop(),
+      });
+    }
+
+    console.log('Instance created.');
+  }
+
+  createInstanceWithCustomHostname();
+  // [END compute_instances_create_custom_hostname]
+}
+
+main(...process.argv.slice(2));
```

Comment:
On an exeptoin, this script will exit with status code `0`, so we may not actually capture the error in tests, instead you can write this:

```js
process.on('unhandledRejection', (err) => {
  console.error(err.message);
  process.exitCode = 1;
});
main(...process.argv.slice(2));
```

Reply:
Sounds reasonable! Added.